tools/topology/topology2/platform/intel/hdmi-generic.conf

```diff
@@ -160,8 +160,10 @@ Object.PCM.pcm [
 		}
 		Object.PCM.pcm_caps.1 {
 			name $HDMI1_PCM_CAPS
-			formats 'S32_LE,S16_LE'
+			formats 'S32_LE,S16_LE,IEC958_SUBFRAME_LE'
 			channels_max 8
+			rate_min 32000
```

Comment:
IIRC 64kHz is not supported by HDaudio and HDMI, is it?

Reply:
yeah, it is not, but can we specify rates or mask rate out with tplg?

Reply:
patch_hmdi.c uses a explicit

````#define SUPPORTED_RATES \
	(SNDRV_PCM_RATE_32000 | SNDRV_PCM_RATE_44100 | SNDRV_PCM_RATE_48000 |\
	SNDRV_PCM_RATE_88200 | SNDRV_PCM_RATE_96000 | SNDRV_PCM_RATE_176400 |\
	 SNDRV_PCM_RATE_192000)
````

btw do we support 44.1kHz with the CHAIN_DMA? I would be surprised if it worked.

Reply:
need to ask @ranj063 and @jsarha if we can have a list of rates in topology instead of min/max?

Reply:
I have tested locally: 44.1, 48, 96 and 192 32bit stereo.

Reply:
I wonder what the buffer sizes are and how we avoid fractional number of samples in the 44.1kHz case?

Reply:
I think this should work:
```
rate_min 0
rate_max 0
rates '32000,44100,48000,88200,96000,176400,192000'
```
Confirmed: it does

Reply:
> I wonder what the buffer sizes are and how we avoid fractional number of samples in the 44.1kHz case?

I have other wip branch to sort out the buffer/period sizes.
The max is the max size allowed, if it is fraction then it is fraction, application does not need to use the whole buffer